### ISSUE TITLE
Accessibility Bug - DAX Logo / Shield button not updating accessibility attributes correctly 

### ISSUE THREAD

**Issue by davidcruzcs** [duckduckgo/iOS#1556]
|||
|-|-|
|***App version***|7.72.6.0|
|***Device model***|iPhone 14 Pro|
|***iOS version***|16.3.1 (20D67)|

***Steps to reproduce the bug***
* Enable VoiceOver
* Open DuckDuckGo and check accessibility hint of the Dax Logo - A11Y Should be OK
* Search and Navigate to any website
* Check Shield Button accessibility hint - A11Y Should be OK
* Go back to the search results
* Check Dax Logo Accessibility Hints - It will be stuck with the Shield hints.

|||
|-|-|
|***Expected behavior***|Once the user returns to the search results, and the DuckDuckLogo shows on the bar, it should have `privacyIconDax` a11y label. and `accessibilityTraits` should be `.image`|
|***Actual behavior***|Once the user returns to the search results, and the DuckDuckLogo shows on the bar, it has `privacyIconShield` a11y label. and `accessibilityTraits` is stuck on `.button`|

https://user-images.githubusercontent.com/10634238/226830900-ee475383-5e2e-4867-a03f-ea9fc2caa75b.mp4

**Comment by davidcruzcs** [comment]
PR: https://github.com/duckduckgo/iOS/pull/1557
